### PR TITLE
clarify distinct pointer type `nil` change

### PR DESCRIPTION
**Pull request by metagn**
refs #20251

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -29,7 +29,7 @@
 - `nimPreviewDotLikeOps` is going to be removed or deprecated.
 
 - The `{.this.}` pragma, deprecated since 0.19, has been removed.
-- `nil` is no longer a valid value for distinct pointer types.
+- The `nil` literal no longer implicitly converts to distinct pointer types.
```
Comment (Varriount) [inline changelog.md]: So, what is the actual change?
- Can a variable with a `nil` value be assigned to a distinct pointer type?
- Can a `nil` literal be used as the assignment expression for a distinct pointer variable?

It sounds like this should be:
````suggestion
- "`nil` literals can no longer be directly assigned to variables or fields of `distinct` pointer types. They must be converted instead.
  ```
  type Foo = distinct ptr int

  # Before:
  var x: Foo = nil
  # After:
  var x: Foo = Foo(nil)
  ```
````

Comment (metagn) [reply changelog.md]: The suggestion is broken, if you want to, you can PR it and I can close this

Comment (Varriount) [reply changelog.md]: Updated.

File: changelog.md [hunk 2]
```diff
@@ -29,7 +29,15 @@
 - `nimPreviewDotLikeOps` is going to be removed or deprecated.
 
 - The `{.this.}` pragma, deprecated since 0.19, has been removed.
-- The `nil` literal no longer implicitly converts to distinct pointer types.
+- "`nil` literals can no longer be directly assigned to variables or fields of `distinct` pointer types. They must be converted instead.
```
Comment (Varriount) [inline changelog.md]:
```suggestion
- `nil` literals can no longer be directly assigned to variables or fields of `distinct` pointer types. They must be converted instead.
```

**Review by Varriount — APPROVED** [review]

File: changelog.md (hunk 2 above)
Comment (metagn) [reply changelog.md]: Already done

**Pull request merged**